### DISCUSSION TITLE
ItemsService actions from an endpoint error : no permission

### DISCUSSION THREAD

**Discussion by zcharef** [directus/directus#17701] [Archive]
**Hello**
I would like to know how can I correctly use the Directus `ItemsService` from an `endpoint` extension ?
My current issue is that my endpoint don't have permission to create an item but shouldn't it be able to do so since the request is coming from a directus extension and not from an external source?

My code looks like this : 

```
...
 router.get("/wine", async (_req, res) => {
      const wineService = new ItemsService('wines', { schema: _req.schema, accountability: _req.accountability });
       await wineService.createOne({
              name: "test",
            })
            res.send({
              status: "OK",
            });
...
```
And the error I am getting : 
```
/Users/.../node_modules/directus/dist/services/authorization.js:368
                throw new exceptions_2.ForbiddenException();
                      ^

ForbiddenException [Error]: You don't have permission to access this.
    at AuthorizationService.validatePayload (/Users/.../node_modules/directus/dist/services/authorization.js:368:23)
    at /Users/.../node_modules/directus/dist/services/items.js:87:46
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  status: 403,
  code: 'FORBIDDEN',
  extensions: {}
}
```
**Best regards,**

**Comment by maxbritto** [comment]
On the request you are using to reach that endpoint, you should add the header `Authorization: Bearer YOUR-USER-AUTH-TOKEN`
Then, inside your endpoint code, your accountability will be filled with this user auth data all all actions will be as if this user had done them. 
If he/she has rights to do them, it will work, otherwise you will get a 403 error.
Note that sometimes, you might also get a 403 error if you just make bad requests (ask for fields that doesn't exist, etc.)
